lib/board.rb

```diff
@@ -0,0 +1,51 @@
+class Board
+  attr_accessor :cells
+
+  def initialize
+    @cells = Array.new(9, " ")
+  end
+
+  def display
+    puts " #{cells[0]} | #{cells[1]} | #{cells[2]} "
+    puts "-----------"
+    puts " #{cells[3]} | #{cells[4]} | #{cells[5]} "
+    puts "-----------"
+    puts " #{cells[6]} | #{cells[7]} | #{cells[8]} "
+  end
+
+  def reset!
+    @cells = Array.new(9, " ")
+  end
+
+  def position(input)
+    cells[input.to_i - 1]
+  end
+
+  def full?
+    cells.all? { |token| token == "X" || token == "O" }
+  end
+
+  def taken?(input)
+    position(input) == "X" || position(input) == "O"
+  end
+
+  def turn_count
+    cells.grep(/[XO]/).size
```

Comment:
Nice use of a regular expression. FYI you can also use [`#count`](http://ruby-doc.org/core-2.2.3/Enumerable.html#method-i-count).